plugins/chat/spec/services/lookup_thread_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::Service::LookupThread do
+  describe Chat::Service::LookupThread::Contract, type: :model do
+    it { is_expected.to validate_presence_of :channel_id }
+    it { is_expected.to validate_presence_of :thread_id }
+  end
+
+  describe ".call" do
+    subject(:result) { described_class.call(params) }
+
+    fab!(:current_user) { Fabricate(:user) }
+    let(:guardian) { Guardian.new(current_user) }
+
+    context "when enable_experimental_chat_threaded_discussions is disabled" do
+      let(:params) { { thread_id: 999, channel_id: 999 } }
+      before { SiteSetting.enable_experimental_chat_threaded_discussions = false }
+      it { is_expected.to fail_a_policy(:threaded_discussions_enabled) }
+    end
+
+    context "when enable_experimental_chat_threaded_discussions is enabled" do
+      before { SiteSetting.enable_experimental_chat_threaded_discussions = true }
+
+      context "when all steps pass" do
+        fab!(:channel) { Fabricate(:chat_channel, threading_enabled: true) }
```

Comment:
Why not define default `fab!` and `params` at the top of the `describe` block and then just change what’s needed for a given example?
This way, when looking at the example, it’s a bit easier to understand what differs from a previous example.

Reply:
This kind of goes against the style @jjaffeux has been doing -- I used to do it the other way as well. We should decide on which way we want to go and stick to it.

Reply:
Im fine trying the let way